### PR TITLE
Implement new preprocessData and types to support new React marks

### LINKED ISSUE #13: Update preprocessData code and types to support new profiling marks
## Steps

- [x] Complete https://github.com/MLH-Fellowship/react/pull/11 to an extent where we can export performance profiles containing marks
- [x] Define new V2 data types to support new marks
- [x] Create preprocessDataV2 to support new data types and new marks

## Acceptance criteria

- A new `ReactProfilerDataV2` and other V2 types should be created to support the new profiling marks.
- preprocessDataV2 should transform a raw profiling JSON into a correct `ReactProfilerDataV2`
- No parsing errors in console on page load
- Neat code, possibly tested (blocked by #14)

### PR DESCRIPTION
**Pull request by taneliang**
## Summary

Implements `preprocessDataV2` and add other V2 types, to support the new user timing marks added in https://github.com/MLH-Fellowship/react/pull/11. `preprocessDataV2` is not being used in app code yet, but tests have been written to ensure that the implementation is mostly correct.

Also added new profiles exported from the Chrome performance tab + versions filtered to contain only `blink.user_timing` events.

Resolves https://github.com/MLH-Fellowship/scheduling-profiler-prototype/issues/13.

### Design

1. `preprocessDataV2` is largely adapted from Brian's original `preprocessData`. The biggest design change I made was to split the React codebase-style pseudo-class/method structure into completely separate functions. Imo this makes it much easier to trace the code as the inputs and outputs of each function is clear, instead of allowing all "methods" in the pseudo-class to mutate some shared variables.

1. `ReactEventV2` is now a disjoint union type. Not sure how useful this will be, but for now this allows us to enforce the presence of some fields that only apply to certain React events. I did not change `ReactMeasureV2` as they all seem to contain the same fields.

Would love to know your thoughts.

### Rollout plan

* Migrate the rest of the app to use the new V2 types and functions. This may take some time.
* Remove the V1 types and functions

## Test plan

`yarn test`

### CODE REVIEW

File: src/constants.js
Comment (kartikcho) [inline src/constants.js:4]: Constants has been moved to `src/canvas` now so you should move this there too.

File: src/types.js
Comment (kartikcho) [inline src/types.js:28]: nit: Let's make this a single comment.

File: src/util/__tests__/preprocessData-test.js
Comment (kartikcho) [inline src/util/__tests__/preprocessData-test.js:6]: Update here too accordingly for the comment in `constants.js`

**Review by kartikcho — CHANGES_REQUESTED** [review]
I didn't check the tests since I'm not very familiar with testing so I'll leave that to @jevakallio , everything else looks good except with some changes left to make.

File: src/types.js
Comment (taneliang) [reply src/types.js:28]: > move this to single comment

What do you mean by this?

File: src/constants.js
Comment (taneliang) [reply src/constants.js:4]: Oh, but this is an app level constant though. I don't think it makes sense to put these in the canvas constants?

Comment (kartikcho) [reply src/constants.js:4]: I was confused about that myself. Maybe we can rename this to `appConstants.js` ? 
But if it's only a single constant in this file, we can add this to the top of `canvas/constant`.

File: src/types.js
Comment (kartikcho) [reply src/types.js:28]: As in a single line comment.

File: src/constants.js
Comment (taneliang) [reply src/constants.js:4]: I think it's fine since they're 2 different domains. I am extremely opposed to putting this in canvas/constants.js as that defeats the purpose of splitting things into folders in the first place, and if we had to rename a file I'd rename the canvas one since these are global constants and are therefore more "authoritative". Either way I think this current set up is okay as the canvas constants file is located within the canvas folder and that folder can be viewed as a self-contained module of its own.

File: src/types.js
Comment (taneliang) [reply src/types.js:28]: That'll be one very long line though (151 chars), which imo is worse than splitting this into 2 lines. I'm breaking lines at the 80 char mark, which is pretty standard for long comments in projects I've seen (and seems more or less standard in the React codebase too). We could also reword this to only take up <80 chars but I'm not sure how we can do that without sacrificing any meaning. Other recommendations welcome, but I'm strongly against changing this to a single long line for readability reasons, and it also causes your editor window to scroll horizontally. The horror.

File: src/constants.js
Comment (kartikcho) [reply src/constants.js:4]: Yeah, that's a valid point. It's good to go from my side!

File: src/types.js
Comment (kartikcho) [reply src/types.js:28]: Split into a comment para if we wanna but that just adds more lines, this is fine I guess

**Review by kartikcho — APPROVED** [review]

File: src/types.js
Comment (taneliang) [reply src/types.js:28]: > Split into a comment para if we wanna but that just adds more lines

What do you mean by this? Do you mean keeping this comment as a 2 line "paragraph"?

Comment (kartikcho) [reply src/types.js:28]: Multi line comments, i.e `/* */`

Comment (taneliang) [reply src/types.js:28]: Ah. That may work but most repos (including React) tend not to use `/* */` for multiline comments. I don't think there's any good technical reason either way, but to be consistent with the React codebase we probably want to stick with the `//` style.

File: src/util/__tests__/preprocessData-test.js
Comment (taneliang) [inline src/util/__tests__/preprocessData-test.js:111]: This array is extremely long and it may be a good idea to put it in a separate JSON file, but I think there's some value in having the data visible when you read the tests, instead of having to poke in another file. Opinions welcome but I think this inlined array is good enough for now.

**Comment by taneliang** [comment]
Merging to unblock https://github.com/MLH-Fellowship/scheduling-profiler-prototype/pull/24. @jevakallio feel free to comment further if you have anything :)

**Pull request merged**
